cuda_core/tests/example_tests/test_basic_examples.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2021-2024 NVIDIA Corporation.  All rights reserved.
+#
+# Please refer to the NVIDIA end user license agreement (EULA) associated
+# with this source code for terms and conditions that govern your use of
+# this software. Any use, reproduction, disclosure, or distribution of
+# this software and related documentation outside the terms of the EULA
+# is strictly prohibited.
+
+# If we have subcategories of examples in the future, this file can be split along those lines
+
+from utils import run_example
+
+def test_basic_examples():
+    run_example("../examples", "saxpy.py")
+    run_example("../examples", "vector_add.py")

```

Comment:
Let's try not to hard-code anything. Forgetting to update test code has been a common reason in past RCCAs that I've seen. Since we've done the heavylifting by referencing nvmath-python, let's do it thoroughly:
```suggestion


samples_path = os.path.join(
    os.path.dirname(__file__), '..', '..', 'examples')
sample_files = glob.glob(samples_path+'**/*.py', recursive=True)


@pytest.mark.parametrize(
    'example', sample_files
)
class TestExamples:

    def test_example(self, example):
        filename = os.path.basename(example)
        run_example(samples_path, example)
```
(plz verify this works :P)